content/en/blog/_posts/protect-mission-critical-pods-priorityclass/index.md

```diff
@@ -0,0 +1,292 @@
+---
+layout: blog
+title: "Protect your mission-critical pods from eviction with PriorityClass"
+date: 2022-11-07
+slug: protect-mission-critical-pods-priorityclass
+description: "Pod priority and preemption help to make sure that mission-critical pods are up in the event of a resource crunch by deciding order of scheduling and eviction."
+---
+
+
+**Author:** Sunny Bhambhani (InfraCloud Technologies)
+
+The adoption of Kubernetes is at a high pace now and many organizations are using this as a de-facto orchestration engine for running their workloads. At this high pace, workloads (and instances of those workloads) are created and deleted frequently.
+
+Therefore, scheduling of pods is crucial in every Kubernetes cluster, and proper scheduling of the pods is the key to running a cluster without issues and making sure that our application pods are up and running. In this article, I will explain some of the use cases around resource management. I'll cover how PriorityClass can help in protecting mission-critical or high-priority pods from getting evicted and making sure that our application pods are up and running and serving traffic.
+
+## Resource Management in Kubernetes 
+
+The control plane consists of multiple components, out of which the scheduler (usually the built-in [kube-scheduler](/docs/concepts/scheduling-eviction/kube-scheduler/)) is one of the components which is responsible for assigning a node to a pod.
+- Whenever a pod is created, it goes to a pending state, and then the scheduler kicks in and determines which node should be the best-suited node for the placement of the new pod.
+  - In the background, the scheduler runs as an infinite loop that looks out for all the pods where the `nodeName` field is null. When it finds one, it starts the process of finding which node should be the best one. Though there are various parameters for it, just to name a few, they are `nodeName`, `nodeSelector`, taints, tolerations, `nodeAffinity`, and the rank of available nodes based on available resources in terms of CPU and memory, etc which decides the pod’s placement.
+- If in-case the scheduler is not able to find any node, the pod remains to be in pending state, which is not an ideal scenario.
+
+The below diagram explains how the request flows in (starting from point number 1 to 4):
+
+{{< figure src=kube-scheduler.svg alt="Whenever there is a new incoming pod, this is how kube-scheduler schedules a pod." title="Kube Scheduler">}}
+
+## Typical Use Cases
+
```

Comment:
This section lacks a preface. You could start by saying that these are a few real-life scenarios where total control over the scheduling and eviction of pods is required. 

I'd recommend rewriting it entirely to incorporate a more generic overview since it delves way too much into specifics.